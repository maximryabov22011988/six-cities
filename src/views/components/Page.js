import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const propTypes = {
  className: PropTypes.string,
  page: PropTypes.string,
  children: PropTypes.element
};

const defaultProps = {
  children: null
};

function Page({ className, page, children }) {
  const classes = classnames(
    className,
    page && `page page--gray page--${page}`
  );
  return <div className={classes}>{children}</div>;
}

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

export default Page;
