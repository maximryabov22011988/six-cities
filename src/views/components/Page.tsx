import * as React from 'react';
import cn from 'classnames';

interface Props {
  children: React.ReactNode,
  className?: string,
  parentClassName?: string,
}

function Page({ className, parentClassName, children }: Props) {
  const classes = cn(className, parentClassName && `page page--gray page--${parentClassName}`);
  return <div className={classes}>{children}</div>;
}

export default Page;
