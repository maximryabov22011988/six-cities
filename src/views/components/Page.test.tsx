import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Page from './Page';

it('Page renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <Page>{null}</Page>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
