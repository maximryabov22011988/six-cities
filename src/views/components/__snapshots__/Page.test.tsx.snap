// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly with mandatory props 1`] = `
<div
  className=""
/>
`;
